crates/router/src/core/routing.rs

```diff
@@ -88,11 +88,13 @@ impl RoutingAlgorithmUpdate {
 pub async fn retrieve_merchant_routing_dictionary(
     state: SessionState,
     merchant_account: domain::MerchantAccount,
+    authentication_profile_id: Option<String>,
     query_params: RoutingRetrieveQuery,
     transaction_type: &enums::TransactionType,
 ) -> RouterResponse<routing_types::RoutingKind> {
     metrics::ROUTING_MERCHANT_DICTIONARY_RETRIEVE.add(&metrics::CONTEXT, 1, &[]);
 
+    core_utils::validate_profile_id_from_auth_layer(authentication_profile_id, &query_params)?;
```

Comment:
is you have `profile_id` in query params, then you can pass it in authentication layer itself to validate it. Or you can handle in handler function. 
Lets try to not have auth in core function whenever possible.